ledger-core/virtual/execute/execute.go

```diff
@@ -720,6 +722,75 @@ func (s *SMExecute) stepSaveNewObject(ctx smachine.ExecutionContext) smachine.St
 	return ctx.Jump(s.stepSendCallResult)
 }
 
+func (s *SMExecute) stepAwaitSMCallSummary(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	syncAccessor, ok := callsummary.GetSummarySMSyncAccessor(ctx, s.execution.Object)
+
+	if ok {
+		var syncLink smachine.SyncLink
+
+		switch syncAccessor.Prepare(func(link *smachine.SyncLink) {
+			syncLink = *link
+		}).TryUse(ctx).GetDecision() {
+		case smachine.NotPassed:
+			return ctx.WaitShared(syncAccessor.SharedDataLink).ThenRepeat()
+		case smachine.Impossible:
+			panic(throw.NotImplemented())
+		case smachine.Passed:
+			// go further
+		default:
+			panic(throw.Impossible())
+		}
+
+		if ctx.AcquireForThisStep(syncLink).IsNotPassed() {
+			return ctx.Sleep().ThenRepeat()
+		}
+	}
+
+	return ctx.Jump(s.stepPublishDataCallSummary)
+}
+
+func (s *SMExecute) stepPublishDataCallSummary(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	callSummaryAccessor, ok := callsummary.GetSummarySMSharedAccessor(ctx, s.pulseSlot.PulseData().PulseNumber)
+
+	if ok {
+		action := func(sharedCallSummary *callsummary.SharedCallSummary) {
+			ref := s.execution.Outgoing
+			requests, ok := sharedCallSummary.Requests.GetObjectsCallResults(s.execution.Object)
```

Comment:
I'd prefer this logic to be hidden inside sharedCallSummary object. 
something like 
```
func (orcr *ObjectsResultCallRegistry) SaveResult(ref, result) bool
```

Reply:
Fixed